frontend/src/services/AuthService.ts

```diff
@@ -1,56 +1,91 @@
-import { Auth } from 'aws-amplify';
-import { readonly, ref } from 'vue';
+import router from "@/router";
+import type { CognitoUserSession } from "amazon-cognito-identity-js";
+import { Auth } from "aws-amplify";
+import { readonly, ref } from "vue";
 
-const FAM_LOGIN_USER = 'famLoginUser'
+const FAM_LOGIN_USER = "famLoginUser";
 
 export interface FamLoginUser {
-    username?: string,
-    idpProvider?: string,
-    roles?: string[],
-    authToken?: CognitoUserSession
+  username?: string;
+  idpProvider?: string;
+  roles?: string[];
+  authToken?: CognitoUserSession;
 }
 
 const state = ref({
-    famLoginUser: localStorage.getItem(FAM_LOGIN_USER)?
-                    JSON.parse(localStorage.getItem(FAM_LOGIN_USER) as string) as (FamLoginUser | undefined | null)
-                    : undefined,
-})
+  famLoginUser: localStorage.getItem(FAM_LOGIN_USER)
+    ? (JSON.parse(localStorage.getItem(FAM_LOGIN_USER) as string) as
+        | FamLoginUser
+        | undefined
+        | null)
+    : undefined,
+});
 
 // functions
 
 function isLoggedIn(): boolean {
-    const loggedIn = !!state.value.famLoginUser?.authToken; // TODO check if token expired later?
-    return loggedIn;
+  const loggedIn = !!state.value.famLoginUser?.authToken; // TODO check if token expired later?
+  return loggedIn;
 }
 
 async function login() {
-    /*
+  /*
         See Aws-Amplify documenation:
         https://docs.amplify.aws/lib/auth/social/q/platform/js/
         https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#identity-pool-federation
     */
-    Auth.federatedSignIn();
+  Auth.federatedSignIn();
 }
 
 async function logout() {
-    Auth.signOut()
-    removeFamUser()
-    console.log("User logged out.")
-    router.push('/')
+  Auth.signOut();
+  removeFamUser();
+  console.log("User logged out.");
+
+  // todo: refactor this part, should get these config variables from backend
+  const env = JSON.parse(window.localStorage.getItem("env_data"));
+  const cognitoConfig = {
+    domain: env?.fam_cognito_domain.value,
+    region: env?.fam_cognito_region.value,
+    userPoolWebClientId: env?.fam_console_web_client_id.value,
+  };
+  const keycloakConfig = {
+    url: "https://dev.oidc.gov.bc.ca/auth",
+    realm: "ichqx89w",
+    siteMinderUrl: "https://logontest7.gov.bc.ca",
+  };
+
+  const postLogoutUrl = window.location.origin;
+  const cognitoLogoutUrl =
+    `${cognitoConfig.domain}.auth.${cognitoConfig.region}.amazoncognito.com/` +
+    `logout?client_id=${cognitoConfig.userPoolWebClientId}&logout_uri=` +
+    postLogoutUrl;
+
+  const keycloakLogoutUrl =
+    keycloakConfig.url +
+    "/realms/" +
+    keycloakConfig.realm +
+    "/protocol/openid-connect/logout?post_logout_redirect_uri=" +
+    cognitoLogoutUrl;
+
+  const siteMinderLogoutUrl =
+    keycloakConfig.siteMinderUrl +
+    "/clp-cgi/logoff.cgi?retnow=1&returl=" +
+    keycloakLogoutUrl;
+
+  router.push(siteMinderLogoutUrl);
```

Comment:
Does the `router.push()` works!? or maybe it does work for router push to trigger browser redirect (I thought it does not work that way previously)?

Reply:
it works, it will redirect to our home page